### PR TITLE
To show badges list of specific address

### LINKED ISSUE #3: badgeOf

### PR DESCRIPTION
**Pull request by geeknees**
fix #3 

# objective
- To show badges list of specific address

# features
- badgesOf
- userBadges
- push userBadges when minted

# spec

```js

  HenkakuBadge
    badgeOf
      ✔ returns minted badges
      ✔ returns empty array without badges

```

### CODE REVIEW

File: contracts/HenkakuBadge.sol [hunk 1]
```diff
@@ -74,6 +77,7 @@ contract HenkakuBadge is ERC1155, Ownable {
         );
         require(success, "TX FAILED");
         _mint(msg.sender, _tokenId, tokenAmount, "");
+        userBadges[msg.sender].push(badges[_tokenId]);
```
Comment (SamuraiT) [inline contracts/HenkakuBadge.sol:80]: good !

File: test/henkakuBadge.ts [hunk 2]
```diff
@@ -1,173 +1,196 @@
...
+        badgeContract.address,
+        ethers.utils.parseUnits('10000', 18)
+      )
+      await badgeContract.mint(1)
+    })
+
+    it('returns badges', async () => {
+      expect(await badgeContract.badgesOf(owner.address)).to.be.eql([
+        [true, false, ethers.utils.parseUnits('100', 18), 'https://example.com']
+      ])
```
Comment (SamuraiT) [inline test/henkakuBadge.ts:193]: can you also add test for non-badge holder

**Comment by geeknees** [comment]
Test case added. I need to add `userBadges[msg.sender].push(badges[_tokenId]);` after the code merge below.

https://github.com/henkaku-center/badge/pull/18

**Review by SamuraiT — APPROVED** [review]

**Review by hourglasshoro — APPROVED** [review]
LGTM!

**Pull request merged**